dotnet/src/InternalUtilities/planning/PlannerInstrumentation.cs

```diff
@@ -160,11 +171,22 @@ internal static partial class PlannerInstrumentation
         Message = "Plan executed successfully.")]
     static partial void LogInvokePlanSuccess(this ILogger logger);
 
-    [LoggerMessage(
-        EventId = 8,
-        Level = LogLevel.Trace, // Sensitive data, logging as trace, disabled by default
-        Message = "Plan result: {PlanResult}")]
-    static partial void LogPlanResult(this ILogger logger, object planResult);
+    private static readonly Action<ILogger, string, Exception?> s_logPlanResult =
+        LoggerMessage.Define<string>(
+            logLevel: LogLevel.Trace,   // Sensitive data, logging as trace, disabled by default
+            eventId: 8,
+            "Plan result: {Result}");
+
+    private static void LogPlanResult(this ILogger logger, object planResult)
+    {
+        if (logger.IsEnabled(LogLevel.Trace))
+        {
+            s_logPlanResult(
+                logger,
+                JsonSerializer.Serialize(planResult),
```

Comment:
What if the plan result is a string? It'd be doubly serialized
```suggestion
                planResult.GetType() == typeof(string) ? planResult : JsonSerializer.Serialize(planResult),
```